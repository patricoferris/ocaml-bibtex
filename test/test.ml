open Bytesrw

let trip filename =
  In_channel.with_open_bin filename @@ fun ic ->
  let reader = Bytes.Reader.of_in_channel ic in
  let bib = Bib.decode ~filename reader in
  Fmt.pr "===== Encoded %s =====\n%!" filename;
  Bib.encode bib (Bytes.Writer.of_out_channel Out_channel.stdout) 

let () =
  trip "test.bib";
  trip "test2.bib"
  

