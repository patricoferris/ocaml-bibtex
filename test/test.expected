===== Encoded test.bib =====
@string{
  hello=world,
  this=is
}
@comment{A little comment to discuss foo bar baz}
@inproceedings{pike1990plan,
  title={Plan 9 from {Bell Labs}},
  author={Pike, Rob and Presotto, Dave and Thompson, Ken and Trickey, Howard and others},
  booktitle={Proceedings of the summer 1990 UKUUG Conference},
  pages={1--9},
  year={1990},
  month=jan,
  organization={London, UK}
}
===== Encoded test2.bib =====
@string{
  hello=world
}
